src/figuresClasses.ts

```diff
@@ -1,19 +1,90 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Colors = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Colors;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    const anySideIsValid = this.a <= 0 || this.b <= 0 || this.c <= 0;
+
+    if (anySideIsValid) {
+      throw new Error('Any side length must be more 0');
+    }
+
+    const oneSideIsValid = this.a >= this.b + this.c
```

Comment:
```suggestion
    const isOneSideValid = this.a >= this.b + this.c
```
According to the name convention, better to prefix boolean variables and function names with "is" or "has"